src/map/battle.c

```diff
@@ -1234,6 +1234,20 @@ static int64 battle_calc_cardfix(int attack_type, struct block_list *src, struct
 							}
 							cardfix = cardfix * (100 - ele_fix_lh) / 100;
 						}
+
+						// Apply bSubDefEle damage reduction.
+						switch (src->type) {
+						case BL_MOB:
+							ele_fix = tsd->sub_def_ele[status_get_element(src)].rate_mob;
```

Comment:
better check return value of status_get_element. probably with assert

Reply:
Fixed.